src/DocumentDB.ChangeFeedProcessor.UnitTests/LeaseManagement/DocumentServiceLeaseTests.cs

```diff
@@ -102,12 +107,121 @@ public void ValidateSerialization_NullFields()
+
+            [JsonProperty("properties")]
+            public Dictionary<string, string> Properties { get; set; } = new Dictionary<string, string>();
+
+            [JsonIgnore]
+            public LeaseAcquireReason AcquireReason { get; set; }
+
+            [JsonProperty("timestamp")]
+            private DateTime? ExplicitTimestamp { get; set; }
+
+            [JsonProperty("_ts")]
+            private long TS { get; set; }
+
+        }
+
+        [Fact]
+        public void ValidateBackwardCompat_OldLeaseFormat()
+        {
+
+            DocumentServiceLeaseV1 originalLease = new DocumentServiceLeaseV1
+            {
+                Id = "id",
+                ETag = "etag",
+                PartitionId = "0",
+                Owner = "owner",
+                ContinuationToken = "continuation",
+                Timestamp = DateTime.Now - TimeSpan.FromSeconds(5),
+                Properties = new Dictionary<string, string> { { "key", "value" } }
+            };
+
+            var serializedV1Lease = JsonConvert.SerializeObject(originalLease);
+            var lease = JsonConvert.DeserializeObject<DocumentServiceLease>(serializedV1Lease);
+
+            Assert.Equal(originalLease.Id, lease.Id);
+            Assert.Equal(null, lease.LeasePartitionKey);
+            Assert.Equal(originalLease.ETag, lease.ETag);
+            Assert.Equal(originalLease.PartitionId, lease.PartitionId);
+            Assert.Equal(originalLease.Owner, lease.Owner);
+            Assert.Equal(originalLease.ContinuationToken, lease.ContinuationToken);
+            Assert.Equal(originalLease.Timestamp, lease.Timestamp);
+            Assert.Equal(originalLease.Properties["key"], lease.Properties["key"]);
+        }
+
+        [Fact]
+        public void ValidateForwardCompat_OldLeaseFormat()
+        {
+
+            DocumentServiceLease originalLease = new DocumentServiceLease
+            {
+                Id = "id",
+                ETag = "etag",
+                PartitionId = "0",
+                Owner = "owner",
+                ContinuationToken = "continuation",
+                Timestamp = DateTime.Now - TimeSpan.FromSeconds(5),
+                Properties = new Dictionary<string, string> { { "key", "value" } }
+            };
+
+            var serializedLease = JsonConvert.SerializeObject(originalLease);
+            var lease = JsonConvert.DeserializeObject<DocumentServiceLeaseV1>(serializedLease);
+
+            Assert.Equal(originalLease.Id, lease.Id);
+            Assert.Equal(originalLease.ETag, lease.ETag);
+            Assert.Equal(originalLease.PartitionId, lease.PartitionId);
+            Assert.Equal(originalLease.Owner, lease.Owner);
+            Assert.Equal(originalLease.ContinuationToken, lease.ContinuationToken);
+            Assert.Equal(originalLease.Timestamp, lease.Timestamp);
+            Assert.Equal(originalLease.Properties["key"], lease.Properties["key"]);
+        }
+        #endregion
```

Comment:
can we also add a test that in JSON serialized payload there is no partitionKey property when DocumentServiceLease doesn't have the value set?